tests/unit/test_loaders.py

```diff
@@ -206,9 +294,116 @@ def test_load_service_model_uses_provided_type_name(self):
     def test_create_loader_parses_data_path(self):
         search_path = os.pathsep.join(['foo', 'bar', 'baz'])
         loader = create_loader(search_path)
-        self.assertIn('foo', loader.search_paths)
-        self.assertIn('bar', loader.search_paths)
-        self.assertIn('baz', loader.search_paths)
+        self.assertIn(pathlib.Path('foo'), loader.search_paths)
+        self.assertIn(pathlib.Path('bar'), loader.search_paths)
+        self.assertIn(pathlib.Path('baz'), loader.search_paths)
+
+    @pytest.mark.skipif(
+        sys.version_info >= (3, 9), reason="Python version >= 3.9"
+    )
+    def test_zips_not_suppored_python_lt_39(self):
+        search_path = os.path.join(
+            os.path.dirname(__file__), 'data', 'Archive.zip', 'foo'
+        )
+        with self.assertRaises(RuntimeError):
+            create_loader(search_path)
```

Comment:
I don't know if we need to explicitly assert this for <3.9. We may want to make the check more explicit that if zipfile.Path doesn't exist, we can't load the zip directory.

Reply:
ok should there be no assertion if it does have `Path`?